lib/contractWrapperBase.ts

```diff
@@ -341,128 +263,129 @@ export interface TransactionReceiptTruffle {
   /**
    * address of the transaction
    */
-  tx: Address;
+  // tx: Address;
 }
 
 export class ArcTransactionResult {
 
+  constructor(
+    /**
+     * The transaction hash
+     */
+    public tx: Hash,
+    /**
+     *  the Truffle contract wrapper
+     */
+    private contract: any) {
+  }
+
   /**
-   * the transaction result to be returned
+   * Returns promise of a mined transaction if it already exists,
+   * converted to a TransactionReceiptTruffle (with readable logs), or null if it doesn't exist.
    */
-  public tx: TransactionReceiptTruffle;
+  public async getTxMined(): Promise<TransactionReceiptTruffle | null> {
+    const tx = await TransactionService.getMinedTransaction(this.tx);
+    return TransactionService.toTxTruffle(tx, this.contract);
+  }
 
-  constructor(tx: TransactionReceiptTruffle) {
-    this.tx = tx;
+  /**
+   * Returns promise of a confirmed transaction if it already exists,
```

Comment:
meaning only return the transaction if it is already confirmed?

Reply:
I have clarified these comments (I hope)